src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ImposeBoundaryConditions.hpp

```diff
@@ -32,11 +33,24 @@ struct Magnitude;
 
 namespace dg {
 namespace Actions {
+namespace BoundaryConditions_detail {
+template <size_t VolumeDim, size_t NumberOfCharSpeeds>
+double min_characteristic_speed(
+    const std::array<DataVector, NumberOfCharSpeeds>& char_speeds) noexcept {
+  std::array<double, NumberOfCharSpeeds> min_speeds{};
+  for (size_t i = 0; i < NumberOfCharSpeeds; ++i) {
+    gsl::at(min_speeds, i) = min(char_speeds.at(i));
+  }
+  return *std::min_element(min_speeds.begin(), min_speeds.end());
+}
+}  // namespace BoundaryConditions_detail
 /// \ingroup ActionsGroup
 /// \ingroup DiscontinuousGalerkinGroup
 /// \brief Packages data on external boundaries for calculating numerical flux.
 /// Computes contributions on the interior side from the volume, and imposes
-/// Dirichlet boundary conditions on the exterior side.
+/// Dirichlet boundary conditions on the exterior side. Optionally, instead do
+/// nothing if all characteristic speeds are outgoing and the system is
+/// nonconservative.
```

Comment:
Can you clarify which char speeds are being evaluated here... the ones from the internal solution evaluated on the exterior of the domain, or the ones from the dirichlet BC solution? What are the implications if they disagree, e.g., what if the internal solution is all-outgoing, but the true BC has inflow? what if the internal solution has inflow, but the true BC is all-outgoing?

Reply:
I'm using the numerical char speeds in the interior to decide what to do here, and I'll clarify the docs that this is the case.

In practice, you'll only have only outgoing char fields on excision boundaries, and once the true solution is only outgoing, but the numerical is incoming, it's already game over. The reverse (the outer boundary numerically has only outgoing char speeds) won't happen unless we move the boundary inward superliminally